dashboard/app/controllers/policy_compliance_controller.rb

```diff
@@ -0,0 +1,31 @@
+class PolicyComplianceController < ApplicationController
+  # GET /policy_compliance/child_account_consent
+  # URL where parents or guardians give consent for their child to have an
+  # account on Code.org
+  # Our child account policy requires that parents give us permission to create
+  # a Code.org account for their child if the child meets the following
+  # criteria:
+  #  * The child is under 13 years old.
+  #  * The child is creating an account using an email address.
+  #  * The child lives in Colorado.
+  def child_account_consent
+    @permission_granted = false
+    token = request.params[:token]
+    return render status: :bad_request if token.blank?
+    #Get parent permission request
+    permission_request = ParentalPermissionRequest.find_by(uuid: token)
+    return render status: :bad_request if permission_request.nil?
+    #Get User
+    user = permission_request.user
+    #Update User
+    if user.child_account_compliance_state != User::ChildAccountCompliance::PERMISSION_GRANTED
+      user.child_account_compliance_state = User::ChildAccountCompliance::PERMISSION_GRANTED
+      user.child_account_compliance_state_last_updated = DateTime.now.new_offset(0)
+      user.save!
+      parent_email = permission_request.parent_email
+      ParentMailer.parent_permission_confirmation(parent_email).deliver_now
```

Comment:
It seems to be nice to revert the user update if something went wrong while sending the `ParentMailer.parent_permission_confirmation` email. If so, it should be inside a transaction.
```suggestion
      User.transaction do
        user.child_account_compliance_state = User::ChildAccountCompliance::PERMISSION_GRANTED
        user.child_account_compliance_state_last_updated = DateTime.now.new_offset(0)
        user.save!
        parent_email = permission_request.parent_email
        ParentMailer.parent_permission_confirmation(parent_email).deliver_now
      end
```

Reply:
The confirmation email being sent is not a critical feature. It is just a "thank you" email.  The most important thing is that the User model is updated since this allows the student to start using their account. 

Thanks for the reminder about transactions though; I will make sure to bring this up in other PR's where the email being sent is critical.